### ISSUE TITLE
Edit Subscription: set Comment ID for the comment author

### ISSUE THREAD

**Issue by IvanRF** [websharks/comment-mail#257]
A user is subscribed to "all comments", and I want to edit the subscription to "only replies to his comment". However, I'm not able to select the comment made by the user. It is not in the list.

User comment id is **1654**. 

Screenshots:
![9-3-2016 9-19-12 p- m-](https://cloud.githubusercontent.com/assets/6751650/13655392/bde78af6-e63d-11e5-9ef6-d293ad854255.png)
![9-3-2016 9-20-55 p- m-](https://cloud.githubusercontent.com/assets/6751650/13655395/c17f7728-e63d-11e5-8ed1-ab0f663c29e1.png)
![9-3-2016 9-25-56 p- m-](https://cloud.githubusercontent.com/assets/6751650/13655396/c3a7f782-e63d-11e5-915e-4d5dfffe7f3c.png)

**Comment by IvanRF** [comment]
I just saw this does not happen for an `unconfirmed` subscription. That is, I can select its own comment when the subscription is `unconfirmed`, but not if it is `subscribed`.

@raamdev is it possible this is a design decision?

**Comment by raamdev** [comment]
I just tested this and I was able to switch a user from All Comments to Replies Only by selecting their own comment (the comment _did_ show up). Also, the user was `subscribed` _not_ `unconfirmed`.

**Comment by IvanRF** [comment]
@raamdev you're right, I just tested it with a different user and his comment is shown.

The issue is different, is that comment id (1654) which is never shown, for any user. I use W3TC, I just cleaned my Database and Object cache in case that was messing things, but that comment is still not showing. I will look in the DB if there is something odd.

**Comment by IvanRF** [comment]
@raamdev I just manually set the comment in the DB, and it is shown in the table but not when I enter Edit Subscription. Could you point me to the DB query in the code for the list in the "Comment ID #" list?

![10-3-2016 10-54-11 a- m-](https://cloud.githubusercontent.com/assets/6751650/13671498/fff52342-e6ae-11e5-9461-b3beedf2bd1c.png)

![10-3-2016 10-55-10 a- m-](https://cloud.githubusercontent.com/assets/6751650/13671506/0a1e9d76-e6af-11e5-9441-5477f358143b.png)

**Comment by raamdev** [comment]
This builds the select option: https://github.com/websharks/comment-mail/blob/160213/comment-mail/includes/classes/utils-markup.php#L433-L540

This fetches comments for a given post ID from the database: https://github.com/websharks/comment-mail/blob/160213/comment-mail/includes/classes/utils-db.php#L586-L676
